great_expectations/data_context/data_context/data_context.py

```diff
@@ -226,18 +226,42 @@ def __init__(
         self,
         context_root_dir: Optional[str] = None,
         runtime_environment: Optional[dict] = None,
+        cloud_mode: bool = False,
+        cloud_base_url: Optional[str] = None,
+        cloud_access_token: Optional[str] = None,
+        cloud_organization_id: Optional[str] = None,
+        # Deprecated as of 0.15.37
         ge_cloud_mode: bool = False,
         ge_cloud_base_url: Optional[str] = None,
         ge_cloud_access_token: Optional[str] = None,
         ge_cloud_organization_id: Optional[str] = None,
     ) -> None:
+        # If any deprecated args deviate from their defaults, we should flag the issue
+        if (
+            ge_cloud_mode
+            is not False  # Due to this being Optional[bool], we can't rely on a falsy check
+            or ge_cloud_base_url
+            or ge_cloud_access_token
+            or ge_cloud_organization_id
+        ):
+            # deprecated-v0.15.37
+            warnings.warn(
```

Comment:
most likely we should sync raising this dep warning with the snippets migration in cloud. Maintain the of statement, but create a Jira task to add the warning in 2023